src/mon/HealthMonitor.cc

```diff
@@ -885,3 +887,79 @@ void HealthMonitor::check_if_msgr2_enabled(health_check_map_t *checks)
     }
   }
 }
+
+void HealthMonitor::check_netsplit(health_check_map_t *checks)
+{
+  // Check for netsplits
+  dout(20) << __func__ << dendl;
+  if (mon.monmap->size() < 3 ||
+    mon.monmap->strategy != MonMap::CONNECTIVITY) {
+    // We need at least 3 monitors to detect netsplits
+    // and the strategy must be CONNECTIVITY.
+    return;
+  }
+  std::set<std::pair<unsigned,unsigned>> bucket_pair = mon.elector.get_netsplit_peer_tracker();
+  std::set<std::pair<std::string,std::string>> sorted_pairs;
+  for (auto i : bucket_pair) {
+    std::string first_mon = mon.monmap->get_name(i.first);
+    std::string second_mon = mon.monmap->get_name(i.second);
+    if (first_mon.empty() || second_mon.empty()) {
+      dout(10) << "Failed to get mon(s) name, either "
+        << first_mon << " or " << second_mon
+        << " was deleted or we have the wrong ranks." << dendl;
+      continue;
+    }
+    std::map<std::string,std::string> first_mon_loc;
+    std::map<std::string,std::string> second_mon_loc;
+    for(auto& mon_info : mon.monmap->mon_info) {
+      if (mon_info.second.name == first_mon) {
+        first_mon_loc = mon_info.second.crush_loc;
+      }
+      else if (mon_info.second.name == second_mon) {
+        second_mon_loc = mon_info.second.crush_loc;
+      }
+    }
+    if (first_mon_loc.empty() || second_mon_loc.empty()) {
+      dout(10) << "Failed to get the mon(s) location, " << first_mon
```

Comment:
Suggest `failed to locate mon(s);` to avoid possessive / plural / singular awkwardness